fcrepo-kernel-modeshape/src/main/java/org/fcrepo/kernel/modeshape/FedoraResourceImpl.java

```diff
@@ -436,6 +436,19 @@ public void updateProperties(final IdentifierConverter<Resource, FedoraResource>
 
         final Collection<IllegalArgumentException> errors = checkInvalidPredicates(request);
 
+        request.getPrefixMapping().getNsPrefixMap().forEach(
+                       (k,v) -> { try {
+                                   LOGGER.info("451: prefix mapping is key -> value: " + k + "->" + v + "\n");
+                  if ( Arrays.asList(getSession().getWorkspace().getNamespaceRegistry().getPrefixes()).contains(k)
+                     && !v.equals(getSession().getWorkspace().getNamespaceRegistry().getURI(k))) {
+                                     LOGGER.info("********451: has exists\n");
+                                    throw new IllegalArgumentException( "prefix has existed");
```

Comment:
Per @ajs6f's earlier comment, please create a new Exception type for this scenario.
For examples see: https://github.com/fcrepo4/fcrepo4/tree/master/fcrepo-kernel-api/src/main/java/org/fcrepo/kernel/api/exception